cpp/log/cert_checker.cc

```diff
@@ -178,12 +179,15 @@ Status CertChecker::CheckIssuerChain(CertChain* chain) const {
 Status CertChecker::CheckPreCertChain(PreCertChain* chain,
                                       string* issuer_key_hash,
                                       string* tbs_certificate) const {
-  if (!chain || !chain->IsLoaded())
+  if (!chain || !chain->IsLoaded()) {
     return Status(util::error::INVALID_ARGUMENT, "invalid certificate chain");
-  Cert::Status status = chain->IsWellFormed();
-  if (status == Cert::FALSE)
+  }
+
+  StatusOr<bool> status = chain->IsWellFormed();
+  if (status.ok() && !status.ValueOrDie()) {
     return Status(util::error::INVALID_ARGUMENT, "prechain not well formed");
-  if (status != Cert::TRUE) {
+  }
+  if (!status.ok() || !status.ValueOrDie()) {
```

Comment:
Didn't do this. It returns a non OK status unless the chain is well formed. I could add the CHECK as an invariant guarantee but I think the if statement is correct.